src/config/config.server.ts

```diff
@@ -0,0 +1,222 @@
+  return Number(variable);
+};
+
+const getEnvironment = (): Environment => {
+  // default to production
+  let environment: Environment = 'production';
+  if (isNotEmpty(ENV('NODE_ENV'))) {
+    switch (ENV('NODE_ENV')) {
+      case 'prod':
+      case 'production':
+        environment = 'production';
+        break;
+      case 'test':
+        environment = 'test';
+        break;
+      case 'dev':
+      case 'development':
+        environment = 'development';
+        break;
+      default:
+        console.warn(`Unknown NODE_ENV ${ENV('NODE_ENV')}. Defaulting to production.`);
+    }
+  }
+
+  return environment;
+};
+
+const getLocalConfigPath = (env: Environment) => {
+  // default to config/config.yml
+  let localConfigPath = join(CONFIG_PATH, 'config.yml');
+
+  if (!fs.existsSync(localConfigPath)) {
+    localConfigPath = join(CONFIG_PATH, 'config.yaml');
+  }
+
+  // determine app config filename variations
+  let envVariations;
+  switch (env) {
+    case 'production':
+      envVariations = ['prod', 'production'];
+      break;
+    case 'test':
+      envVariations = ['test'];
+      break;
+    case 'development':
+    default:
+      envVariations = ['dev', 'development']
+  }
+
+  // check if any environment variations of app config exist
+  for (const envVariation of envVariations) {
+    let envLocalConfigPath = join(CONFIG_PATH, `config.${envVariation}.yml`);
+    if (fs.existsSync(envLocalConfigPath)) {
+      localConfigPath = envLocalConfigPath;
+      break;
+    } else {
+      envLocalConfigPath = join(CONFIG_PATH, `config.${envVariation}.yaml`);
+      if (fs.existsSync(envLocalConfigPath)) {
+        localConfigPath = envLocalConfigPath;
+        break;
+      }
+    }
+  }
+
+  return localConfigPath;
+};
+
+const overrideWithConfig = (config: Config, pathToConfig: string) => {
+  try {
+    console.log(`Overriding app config with ${pathToConfig}`);
+    const externalConfig = fs.readFileSync(pathToConfig, 'utf8');
+    mergeConfig(config, yaml.load(externalConfig));
+  } catch (err) {
+    console.error(err);
+  }
+};
+
+const overrideWithEnvironment = (config: Config, key: string = '') => {
+  for (const property in config) {
+    const variable = `${key}${isNotEmpty(key) ? '_' : ''}${property.toUpperCase()}`;
+    const innerConfig = config[property];
+    if (isNotEmpty(innerConfig)) {
+      if (typeof innerConfig === 'object') {
+        overrideWithEnvironment(innerConfig, variable);
+      } else {
+        const value = ENV(variable, true);
+        if (isNotEmpty(value)) {
+          console.log(`Applying environment variable ${DSPACE(variable)} with value ${value}`);
+          switch (typeof innerConfig) {
+            case 'number':
+              config[property] = getNumberFromString(value);
+              break;
+            case 'boolean':
+              config[property] = getBooleanFromString(value);
+              break;
+            case 'string':
+              config[property] = value;
```

Comment:
Looks like you forgot the `break` here. 
Currently you always get the `default` warning below for any string

Reply:
Thanks. Will update.

Reply:
Added.